### PR TITLE
[FLINK-11121][license] Check and update licensing notes for Aliyun FS

### PR DESCRIPTION
**Pull request by StefanRRichter**
Licensing files for Aliyun File System, as introduced with FLINK-10865.

### CODE REVIEW

**Comment by StefanRRichter** [comment]
CC @zentol

**Comment by zentol** [comment]
@flinkbot approve description
@flinkbot approve consensus

**Comment by zentol** [comment]
@flinkbot approve architecture

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+- com.nimbusds:nimbus-jose-jwt:4.41.1
+- commons-beanutils:commons-beanutils:1.9.3
+- commons-cli:commons-cli:1.3.1
+- commons-codec:commons-codec:1.10
+- commons-collections:commons-collections:3.2.2
+- commons-io:commons-io:2.4
+- commons-lang:commons-lang:2.6
+- commons-logging:commons-logging:1.1.3
+- commons-net:commons-net:3.6
+- net.bytebuddy:byte-buddy-agent:1.8.15
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE]: this is a test dependency, hence not bundled in the jar, hence doesn't have to be listed in NOTICE

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+- net.bytebuddy:byte-buddy:1.8.15
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE]: test dependency

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+- org.codehaus.jettison:jettison:1.1
+- org.eclipse.jetty:jetty-http:9.3.19.v20170502
+- org.eclipse.jetty:jetty-io:9.3.19.v20170502
+- org.eclipse.jetty:jetty-security:9.3.19.v20170502
+- org.eclipse.jetty:jetty-server:9.3.19.v20170502
+- org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
+- org.eclipse.jetty:jetty-util:9.3.19.v20170502
+- org.eclipse.jetty:jetty-webapp:9.3.19.v20170502
+- org.eclipse.jetty:jetty-xml:9.3.19.v20170502
+- org.javassist:javassist:3.19.0-GA
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE]: test dependency

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+- com.sun.jersey:jersey-servlet:1.19
+- com.sun.xml.bind:jaxb-impl:2.2.3-1
+- javax.servlet.jsp:jsp-api:2.1
+- javax.servlet:javax.servlet-api:3.1.0
+- javax.xml.bind:jaxb-api:2.2.2
+
+This project bundles the following dependencies under the MIT/X11 license.
+See bundled license files for details.
+
+- org.slf4j:slf4j-log4j12:1.7.15
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE]: this is a provided dependency, hence not bundled in the jar, hence doesn't have to be listed in NOTICE

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+This project bundles the following dependencies under the Go License (https://golang.org/LICENSE).
+See bundled license files for details.
+
+- com.google.re2j:re2j:1.1
+
+ Copyright (C) 2000-2004 Jason Hunter & Brett McLaughlin.
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE]: The license is an extension of bsd-3, so I would move the license itself into `LICENSE.jdom` and add
```
This project bundles the following dependencies under a BSD-like license.
See bundled license files for details.
```
above the dependency entry.

Alternatively you also refer to it as "the JDOM license".

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+ This software consists of voluntary contributions made by many
+ individuals on behalf of the JDOM Project and was originally
+ created by Jason Hunter <jhunter_AT_jdom_DOT_org> and
+ Brett McLaughlin <brett_AT_jdom_DOT_org>.  For more information
+ on the JDOM Project, please see <http://www.jdom.org/>.
+
+- org.jdom:jdom:1.1
+
+This project bundles the following dependencies under the following license.
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE]: Move to `LICENSE.json`, refer to as `JSON license`

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/licenses/LICENSE.slf4j [hunk 7]
```diff
@@ -0,0 +1,21 @@
+Copyright (c) 2004-2017 QOS.ch
+ All rights reserved.
+
+ Permission is hereby granted, free  of charge, to any person obtaining
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/licenses/LICENSE.slf4j]: the license is't needed, see comment about NOTICE file.

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE [hunk 8]
```diff
@@ -0,0 +1,205 @@
+flink-oos-fs-hadoop
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/NOTICE]: typo: oos -> oss

**Comment by zentol** [comment]
@flinkbot approve architecture

**Comment by StefanRRichter** [comment]
@zentol thanks for the quick review. I updated the PR to address all comments. Please take another look.

**Comment by zentol** [comment]
Needs a rebase now that #7599 was merged.

**Comment by StefanRRichter** [comment]
rebased

**Comment by StefanRRichter** [comment]
Thanks for the review @zentol! I addressed your final comment, merging the PR.

**Pull request merged**